paddle/fluid/operators/conv_op.cc

```diff
@@ -48,7 +48,11 @@ void ConvOp::InferShape(framework::InferShapeContext* ctx) const {
   int groups = ctx->Attrs().Get<int>("groups");
   std::vector<int> dilations = ctx->Attrs().Get<std::vector<int>>("dilations");
   const std::string data_format = ctx->Attrs().Get<std::string>("data_format");
-  const bool channel_last = (data_format == "NHWC" || data_format == "NDHWC");
+
+  // MKL-DNN Kernels are using NCHW order of dims description
+  // so we ignore data_format consideration for MKL-DNN kernel
+  const bool channel_last = (this->IsMKLDNNType() == false) &&
+                            (data_format == "NHWC" || data_format == "NDHWC");
```

Comment:
Since this PR only supports LRN and Pool2d (FWD) NHWC, please don't change other Ops.

Reply:
missed that one. OK